src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -202,7 +204,14 @@ func (r *StatefulSetResource) shouldUpdate(
 		return false, err
 	}
 
-	return !patchResult.IsEmpty() || upgrading, nil
+	anyDepsUpdated := false
+	for _, t := range r.updateTriggers {
+		anyDepsUpdated = anyDepsUpdated && r.updatedResources[t]
```

Comment:
if it's any, then this needs to be `anyDepsUpdated || r.updatedResources[t]`

Reply:
Omg, you're totally right. My bad!